app/controllers/api/functions.php

```diff
@@ -840,6 +889,21 @@
             $live = false;
         }
 
+        $spec = Config::getParam('runtime-specifications')[$specification] ?? [];
+
+        // Enforce Cold Start if spec limits change.
+        if ($function->getAttribute('specification') !== $specification && !empty($function->getAttribute('deployment'))) {
+            $executor = new Executor(App::getEnv('_APP_EXECUTOR_HOST'));
```

Comment:
Do we have a dependency for this we can inject instead of instantiating inline?

Reply:
No, we do not currently have a dependency for this